### PR TITLE
[Diagnostics] Add a fix-it for try instead of throws in function decl…

### PR DESCRIPTION
**Pull request by vguerra**
…s and types.

Sometimes one would misstype `throws` for `try` so we provide a fix-it for that.
It might happen in two cases:

In function declarations:
```swift
func foo() try {}
```

In function types:
```swift
let f = () try -> Int
```

Resolves [SR-11574](https://bugs.swift.org/browse/SR-11574).

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by vguerra** [comment]
My bad! .. PR was not compiling. I pushed 59e8f20 with the fix.

@CodaFi, could you pls trigger tests again? Thanks!

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Comment by rintaro** [comment]
@vguerra Sorry. The conflict is due to https://github.com/apple/swift/pull/27675 . Could you update the PR?

**Comment by vguerra** [comment]
@rintaro no worries. updated.

File: lib/Parse/ParsePattern.cpp [hunk 1]
```diff
@@ -786,6 +786,10 @@ Parser::parseFunctionSignature(Identifier SimpleName,
     throwsLoc = consumeToken();
     diagnose(throwsLoc, diag::throw_in_function_type)
       .fixItReplace(throwsLoc, "throws");
+  } else if (Tok.is(tok::kw_try)) {
+    SourceLoc tryLoc = consumeToken();
+    diagnose(tryLoc, diag::throw_in_function_type)
+        .fixItReplace(tryLoc, "throws");
```
Comment (rintaro) [inline lib/Parse/ParsePattern.cpp]: I think we should pretend that the `throws` is correctly written. Does just modifying the condition of the previous block to `if (Tok.isAny(tok::kw_throw, kw_try))` work?

Comment (vguerra) [reply lib/Parse/ParsePattern.cpp]: yes, setting `throwsLoc` does work. I pushed 965605732e5ce44b8c55ebc5f85f1ec279636621

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by rintaro — APPROVED** [review]
Thank you Victor!

**Pull request merged**